src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -182,6 +191,82 @@ private void PopulateAssemblies()
             }
         }
 
+        /// <summary>
+        /// Update FileDefinitions and FileDependencies to pretend that certain Reference items
+        /// explicitly added by a .targets file in the NETStandard.Library are actually normal
+        /// package items. This allows them to show up properly under the "SDK" node in Solution
+        /// Explorer, rather than just under the "Assemblies" node.
+        ///
+        /// This is not meant to be a general mechanism for injecting files that can't be handled
+        /// through the normal NuGet package mechanisms. For that, package authors should be
+        /// creating FileDefinition and FileDependency items instead of Reference items.
+        /// </summary>
+        private void PopulatePreResolvedNetStandardAssemblies()
+        {
+            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            foreach (var reference in References)
+            {
+                var packageName = reference.GetMetadata("NuGetPackageId");
+                var packageVersion = reference.GetMetadata("NuGetPackageVersion");
+
+                // This is not a "pre-resolved" assembly; skip it.
+                if (packageName == null || packageVersion == null)
+                {
+                    continue;
+                }
+
+                // This is meant only as a workaround for an issue in NETStandard.Library 2.0.*.
+                // Other packages should use the more general mechanism of creating
+                // FileDefinition and FileDependency items rather than Reference items.
+                if (packageName != "NETStandard.Library")
+                {
+                    continue;
+                }
+
+                var versionComponents = packageVersion.Split('.');
+                if (versionComponents[0] != "2" ||
```

Comment:
That's a bad assumption.  You wouldn't need to assume if you asked the folks who owned the package.  Now we need to come ask for a fix when we make NETStandard.Library 2.1?